src/wp-includes/option.php

```diff
@@ -823,7 +818,15 @@ function update_option( $option, $value, $autoload = null ) {
 	);
 
 	if ( null !== $autoload ) {
-		$update_args['autoload'] = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+		$autoload                = get_autoload_value( $autoload );
+		$update_args['autoload'] = $autoload;
+	} else {
+		$raw_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s LIMIT 1", $option ) );
```

Comment:
We should only change `default` values. If an option is saved as yes or no, this should be considers that the developer picked these values.